code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1414,3 +1415,315 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
+		/obj/item/weapon/reagent_containers/food/snacks/grown/soybeans = list("soymilk" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = list("ketchup" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/corn = list("cornoil" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/wheat = list("flour" = -5),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/ricestalk = list("rice" = -5),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/cherries = list("cherryjelly" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/plastellium = list("plasticide" = 5),
+		/obj/item/seeds = list("blackpepper" = 5),
+		/obj/item/weapon/rocksliver = list("ground_rock" = 30),
+
+		//!Put all recipes above this!
+		/obj/item/weapon/reagent_containers/food = list()
+	)
+
+	var/list/juice_items = list (
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = list("tomatojuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/carrot = list("carrotjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/berries = list("berryjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/banana = list("banana" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/potato = list("potato" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lemon = list("lemonjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/orange = list("orangejuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lime = list("limejuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/watermelonslice = list("watermelonjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/poisonberries = list("poisonberryjuice" = 0),
+	)
+
+
+	var/obj/item/crushable = null
+
+
+/obj/item/weapon/reagent_containers/mortar/afterattack(obj/target, mob/user , flag)
+	if(target.is_open_container() && target.reagents) //Something like a glass. Player probably wants to transfer TO it.
+		if(!reagents.total_volume)
+			user << "<span class='warning'>[src] is empty.</span>"
+			return
+
+		if(target.reagents.total_volume >= target.reagents.maximum_volume)
+			user << "<span class='warning'>[target] is full.</span>"
+			return
+
+		var/trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
+		user << "<span class='notice'>You transfer [trans] units of the solution to [target].</span>"
+
+/obj/item/weapon/reagent_containers/mortar/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	if (isscrewdriver(O))
+		if(crushable)
+			crushable.loc = src.loc
+		new /obj/item/stack/sheet/metal(user.loc)
+		new /obj/item/trash/bowl(user.loc)
+		return
+	if (crushable)
+		user << "<span class ='warning'> There's already something inside!</span>"
+		return 1
+	if (!is_type_in_list(O, blend_items) && !is_type_in_list(O, juice_items))
+		user << "<span class ='warning'> You can't grind that!</span>"
+		return ..()
+
+	user.before_take_item(O)

```

Comment:
It (and the "updateUsrDialog" below) appears to be a typo related to SERIOUS HEAVY DUTY code stealing from the grinder.